### ISSUE TITLE
Cannot catch double import in TYPE_CHECKING block

### ISSUE THREAD

**Issue by beskep** [astral-sh/ruff#12270]
command: `ruff check --select TCH,I,F`
version: 0.5.1

code: 
```python
from decimal import Decimal  # I001: import block un-sorted
from decimal import Decimal  # F811: redefinition
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from decimal import Decimal  # reimport, but no error


def fn():
    value: Decimal  # noqa: F842
```

I often import the same bunch of modules by snippet after TCH creates TYPE_CHECKING block.
ruff --fix deletes the second line above and keeps the first line.

**Comment by charliermarsh** [comment]
Just to be clear, which line do you think should be flagged as an error, and which should be removed? (And similarly, what if `Decimal` is used outside of a type annotation -- what would you expect?)

**Comment by beskep** [comment]
Removing the first and second lines would be perfect.

Or maybe flag later imports (lines 2 and 6) and remove line 2, 5 and 6.
Then the `TCH` will handle `TYPE_CHECKING` block.

**Comment by ntBre** [comment]
I believe this should be resolved in the next release by the changes in https://github.com/astral-sh/ruff/pull/22560!
